ggmolvis/properties/materials.py

```diff
@@ -35,7 +35,21 @@ def set_material(self, material_name):
     def _apply_to(self, obj, frame: int = 0):
         obj.data.materials.append(self.material)
     
-    def _modify_material(self, material, frame: int = 0):
+    def _modify_material(self, material, frame: int = 0,
+                         color=None):
+        color_dict = {"violet": (0.58, 0.0, 0.827, 1.0),
+                      "green": (0, 0.9, 0.0, 1.0),
+                      "red": (0.9, 0.0, 0.0, 1.0),
+                      "blue": (0.0, 0.0, 0.9, 1.0),
+                      "black": (0.0, 0.0, 0.0, 1.0),
+                      "default": (0.0, 0.0, 0.0, 1.0),
+                      }
+        if color is not None:
+            material.node_tree.nodes.clear()
+            principled = material.node_tree.nodes.new("ShaderNodeBsdfPrincipled")
+            principled.inputs["Base Color"].default_value = color_dict[color]
```

Comment:
We may actually want to link back to the vertices/mesh like in gh-23 wtih something like the attribute node over there set up:

`attribute = material.node_tree.nodes.new("ShaderNodeAttribute")`

I think I was seeing some failures with that approach, but my workflow is frequently confounded by https://github.com/yuxuanzhuang/ggmolvis/issues/26#issuecomment-2719787101.

Reply:
In gh-23 Brady indicated:

> setting the color attributes on the mesh rather than through nodes.

I'm not sure we can avoid using nodes of any kind completely, but I suspect he might mean linking back to the mesh here?

Reply:
On the other hand, it is pretty important that I have short-term color setting capability for current deliverables.

If we link back to the vertices/mesh will it be trivial to control *those* colors from `ggmolvis` without a bunch more refactoring?